internal/corazawaf/rule.go

```diff
@@ -175,6 +206,32 @@ func (r *Rule) doEvaluate(phase types.RulePhase, tx *Transaction, cache map[tran
 		rid = r.ParentID_
 	}
 
+	// TODO(anuraaga): This is effectively lazily computing the min phase of a rule with chain the first
+	// time we evaluate the rule. Instead, we should do this at parse time, but this will require a
+	// large-ish refactoring of the parser, which adds parent rules to a rule group before preparing
+	// the child rules. In the meantime, only evaluating this once should allow performance to be fine.
+	//
+	// chainMinPhase is the minimum phase among all the rules in which the chained rule may match.
+	// We evaluate the min possible phase for each rule in the chain and we take the minimum in common
+	// If we reached this point, it means that the parent rule already reached its min phase.
+	if multiphaseEvaluation && r.ParentID_ == 0 && r.HasChain && r.chainMinPhase == types.PhaseUnknown {
```

Comment:
Can all this logic go into another function and let the `if` only checking if multiphase is true?